server/src/test/java/io/crate/integrationtests/PrivilegesIntegrationTest.java

```diff
@@ -308,6 +308,42 @@ public void testPrivilegeIsSwappedWithSwapTable() {
         assertThat(response.rowCount(), is(0L));
     }
 
+    @Test
+    public void test_alter_cluster_swap_table_works_for_nonsueruser() {
+        executeAsSuperuser("create table doc.t1 (x int)");
+        executeAsSuperuser("create table doc.t2 (x int)");
+        executeAsSuperuser("grant dql on table doc.t1 to " + TEST_USERNAME);
+
+        //Before swapping - can access t1 but cannot access t2
+
+        // Test user can access table t1
+        executeAs("select * from doc.t1", TEST_USERNAME);
+        assertThat(response.rowCount(), is(0L)); // Test user can access table t2
+
+        // Test user cannot access table t2
+        assertThrows(
+            () -> executeAs("select * from doc.t2", TEST_USERNAME),
+            isSQLError(is("Schema 'doc' unknown"), INTERNAL_ERROR, NOT_FOUND, 4045)
+        );
+
+        // User gets AL privileges and swaps tables
+        executeAsSuperuser("grant AL to " + TEST_USERNAME);
+        executeAs("alter cluster swap table doc.t1 to doc.t2 with (drop_source = true)", TEST_USERNAME);
+
+        //After swapping - cannot access t1 but can access t2
+
+        // Test user cannot access table t1
+       // executeAs("select * from doc.t1", TEST_USERNAME);
+        assertThrows(
+            () -> executeAs("select * from doc.t1", TEST_USERNAME),
+            isSQLError(is("Relation 'doc.t1' unknown"), INTERNAL_ERROR, NOT_FOUND, 4041)
```

Comment:
I have a concern here that first "cannot access" check was Schema unknown with code 4045 and after the swap it was relation unknown  - is it expected behavior? I guess it's because user knows already about the schema after getting DQL on t1 but pointed out that assertion difference just in case.

Reply:
As this test should afaik only test that the swap operation is allowed for users with `AL` privileges, do we really need all the other assertions checking the table privileges? I'd suggest to only test the swap privileges.

Reply:
Thanks for the explanation! Removed extra assertions